src/cargo/ops/cargo_run.rs

```diff
@@ -46,13 +46,31 @@ pub fn run(
             // this will be verified in cargo_compile
         }
     }
+
+    if bins.len() == 1 {
+        let bin = bins.first().unwrap();
```

Comment:
This is safe right?

Reply:
Yep, this is ok! You could use something like `let (name, kind) = bins.first().unwrap()` to avoid `.0` and `.1`.